src/main/java/piggy/Ui.java

```diff
@@ -0,0 +1,129 @@
+package piggy;
+
+import java.util.List;
+import java.util.Scanner;
+
+import piggy.task.Task;
+
+/**
+ * Manages the user-interaction UI.
+ */
+class Ui {
+    private final Scanner sc;
+
+    /**
+     * Creates a new UI object. Uses STDIN for input and STDOUT for output.
+     */
+    Ui() {
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * Shows the welcome message on the output.
+     */
+    void showWelcome() {
+        System.out.println("Hello! I'm Piggy");
+        System.out.println("What can I oink for you?");
+        System.out.println("Use the following format for datetime: yyyy-MM-dd HH:mm");
+    }
+
+    /**
+     * Shows the bye message on the output and stops receiving user input.
+     */
+    void showBye() {
+        System.out.println("Bye. Hope to oink you again soon!");
+        sc.close();
+    }
+
+    /**
+     * Shows the list of tasks on the output.
+     *
+     * @param tasks The list of tasks to show.
+     */
+    void showTaskList(List<Task> tasks) {
+        System.out.println("Here are the tasks in your list:");
+        showTasks(tasks);
+    }
+
+    private void showTasks(List<Task> tasks) {
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.printf("%d.%s\n", i + 1, tasks.get(i));
+        }
+    }
+
+    /**
+     * Show that a task has been added along with the remaining tasks.
+     *
+     * @param task             The added task to show.
+     * @param noRemainingTasks The number of remaining tasks.
+     */
+    void showTaskAdded(Task task, int noRemainingTasks) {
```

Comment:
Consider renaming this method to `showAddTask` or `showAddedTask`? The ordering of the words makes the method name seems unclear to me.